File: back-end/src/main/java/com/ta2khu75/quiz/service/ExamCategoryService.java
package com.ta2khu75.quiz.service;

import java.util.List;

import com.ta2khu75.quiz.model.request.ExamCategoryRequest;
import com.ta2khu75.quiz.model.response.ExamCategoryResponse;
import com.ta2khu75.quiz.service.base.CrudService;

public interface ExamCategoryService extends CrudService<ExamCategoryRequest, ExamCategoryResponse, Long> {
	List<ExamCategoryResponse> readAll();
}

File: front-end/src/service/ExamCategoryService.ts
import ApiResponse from "../model/response/ApiResponse";
import instance from "../util/apiInstance";
const basePath = "exam-category";
export default class ExamCategoryService {
    static readAll(): Promise<ApiResponse<ExamCategoryResponse[]>> {
        return instance.get(basePath);
    }
    static create(category: ExamCategoryRequest): Promise<ApiResponse<ExamCategoryResponse>> {
        return instance.post(basePath, category);
    }
    static delete(id: number): Promise<ApiResponse<void>> {
        return instance.delete(`${basePath}/${id}`);
    }
    static update(id: number, category: ExamCategoryRequest): Promise<ApiResponse<ExamCategoryResponse>> {
        return instance.put(`${basePath}/${id}`, category);
    }
}